sormas-backend/src/test/java/de/symeda/sormas/backend/sample/PathogenTestFacadeEjbTest.java

```diff
@@ -1,45 +1,155 @@
+        final CaseDataDto caze = creator.createCase(user.toReference(), person.toReference(), rdcf);
+        final ContactDto contact = creator.createContact(user.toReference(), person.toReference(), caze);
+        final VisitDto visit = creator.createVisit(caze.getDisease(), person.toReference());
+        final SampleDto sample = creator.createSample(contact.toReference(), new Date(), new Date(), user.toReference(),
+                SampleMaterial.BLOOD, rdcf.facility);
+
+        final CaseDataDto caseDataDto = CaseDataDto.buildFromContact(contact, visit);
+        caseDataDto.setRegion(new RegionReferenceDto(rdcf.region.getUuid()));
+        caseDataDto.setDistrict(new DistrictReferenceDto(rdcf.district.getUuid()));
+        caseDataDto.setHealthFacility(new FacilityReferenceDto(rdcf.facility.getUuid()));
+        caseDataDto.setReportingUser(user.toReference());
+        final CaseDataDto caseConvertedFromContact = getCaseFacade().saveCase(caseDataDto);
+
+        getCaseFacade().setSampleAssociations(contact.toReference(), caseConvertedFromContact.toReference());
+
+        final PathogenTestDto newPathogenTest = createPathogenTestDto(rdcf, user, sample, caseConvertedFromContact.getDisease());
+
+        testSaveAndUpdatePathogenTest(newPathogenTest);
+    }
+
+    private void testSaveAndUpdatePathogenTest(PathogenTestDto newPathogenTest) {
+        final PathogenTestDto savedPathogenTest = getPathogenTestFacade().savePathogenTest(newPathogenTest);
+        assertNotNull(savedPathogenTest);
+        assertEquals(Disease.EVD, savedPathogenTest.getTestedDisease());
+        assertEquals(PathogenTestType.ISOLATION, savedPathogenTest.getTestType());
+        assertEquals(PathogenTestResultType.PENDING, savedPathogenTest.getTestResult());
+        assertEquals("all bad!", savedPathogenTest.getTestResultText());
+        assertEquals(testDateTime, savedPathogenTest.getTestDateTime());
+        assertFalse(savedPathogenTest.getTestResultVerified());
+
+        savedPathogenTest.setTestResultVerified(true);
+        savedPathogenTest.setTestResultText("all good!");
+        savedPathogenTest.setTestType(PathogenTestType.OTHER);
+        savedPathogenTest.setTestResult(PathogenTestResultType.POSITIVE);
+
+        final PathogenTestDto updatedPathogen = getPathogenTestFacade().savePathogenTest(savedPathogenTest);
+        assertNotNull(updatedPathogen);
+        assertEquals(Disease.EVD, updatedPathogen.getTestedDisease());
+        assertEquals(PathogenTestType.OTHER, updatedPathogen.getTestType());
+        assertEquals(PathogenTestResultType.POSITIVE, updatedPathogen.getTestResult());
+        assertEquals("all good!", updatedPathogen.getTestResultText());
+        assertEquals(testDateTime, updatedPathogen.getTestDateTime());
+        assertTrue(updatedPathogen.getTestResultVerified());
+    }
+
+    private PathogenTestDto createPathogenTestDto(RDCFEntities rdcf, UserDto user, SampleDto sample, Disease disease) {
```

Comment:
Move this to TestDataCreator and change the method name to "buildPathogenTestDto" to hint at it not being saved to the database automatically